### DISCUSSION TITLE
How to reduce the memory footprint of cloud-hypervisor

### DISCUSSION THREAD

**Discussion by ljrcore** [cloud-hypervisor/cloud-hypervisor#4961] [General]
We obtained the memory footprint overhead of CLH  with the vmm_memory tool.
```
SKIP: /memfd:ch_ram has size 266272 kB
  0    kB -- [[vsyscall]]
  0    kB -- [[vvar]]
  4    kB -- [[vdso]]
  4    kB -- [/dev/zero]
  8    kB -- [anon_inode:kvm-vcpu:0]
  20   kB -- [/usr/lib64/libdl-2.28.so]
  24   kB -- [/usr/lib64/libcap-ng.so.0.0.0]
  24   kB -- [[stack]]
  76   kB -- [/usr/lib64/libseccomp.so.2.5.1]
  84   kB -- [[heap]]
  96   kB -- [/usr/lib64/libgcc_s-8-20210514.so.1]
  112  kB -- [/usr/lib64/libpthread-2.28.so]
  188  kB -- [/usr/lib64/ld-2.28.so]
  320  kB -- [/usr/lib64/libm-2.28.so]
  824  kB -- []
  1388 kB -- [/usr/lib64/libc-2.28.so]
  3676 kB -- [/root/cloudliang/cloud-hypervisor/target/release/cloud-hypervisor]
  108384kB -- [/root/cloudliang/focal-server-cloudimg-amd64.raw]
Total Overhead: 115232 kB
```
This data calculates the CLH memory footprint to be approximately 6.688M. Sometimes the CLH memory footprint can reach 8M.

How should we analyze the memory footprint in a more granular way? Besides image, how should we reduce the CLH memory footprint?

release v28.0

Thanks

**Comment by liuw** [comment]
What's your target?

Heap + stack is only 108KB. There is another 824KB which I'm not sure about.

Shared objects and CH itself are going to be shared among many processes. They aren't a concern.

If you want to reduce the image size, use `panic=abort`.
